fvm/evm/types/result.go

```diff
@@ -0,0 +1,54 @@
+package types
+
+import (
+	gethCommon "github.com/ethereum/go-ethereum/common"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+)
+
+// tx type 255 is used for direct calls from bridged accounts
+var DirectCallTxType = uint8(255)
+
+// Result captures the result of an interaction to the emulator
+// it could be the out put of a direct call or output of running an
+// evm transaction.
+// Its more comprehensive than typical evm receipt, usually
+// the receipt generation requires some extra calculation (e.g. Deployed contract address)
+// but we take a different apporach here and include more data so that
+// it requires less work for anyone who tracks and consume results.
+type Result struct {
+	// a boolean that is set to false if the execution has failed (non-fatal)
+	Failed bool
+	// type of transaction defined by the evm package
+	// see DirectCallTxType as extra type we added type for direct calls.
+	TxType uint8
+	// total gas consumed during an opeartion
+	GasConsumed uint64
+	// the root hash of the state after execution
+	StateRootHash gethCommon.Hash
+	// the address where the contract is deployed (if any)
+	DeployedContractAddress Address
+	// returned value from a function call
+	ReturnedValue []byte
```

Comment:
Should this be of type `Data`?

Reply:
It could be, Data is right now used for input data, but both represent the same byte slice, going to change it in the future PRs since it would mostly impact the code that is setting the value for result.